source/opt/instruction.cpp

```diff
@@ -50,6 +50,20 @@ Instruction::Instruction(const spv_parsed_instruction_t& inst,
   }
 }
 
+Instruction::Instruction(SpvOp op, uint32_t type_id, uint32_t result_id,
+                         const std::vector<Operand>& in_operands)
+    : opcode_(op), type_id_(type_id), result_id_(result_id), operands_() {
+  if (type_id_ != 0) {
+    operands_.emplace_back(
+        Operand(spv_operand_type_t::SPV_OPERAND_TYPE_TYPE_ID, {type_id_}));
```

Comment:
Hmm... I think this is still constructing a temporary object and then constructed? Considering `emplace_back(SPV_OPERAND_TYPE_TYPE_ID, std::vector<uint32_t>{type_id})`?